src/components/FilterView.js

```diff
@@ -0,0 +1,119 @@
+import { useState } from "react";
+
+
+const idealFor  = [ 'Boys', 'Girls', 'Men', 'Unisex', 'Women' ];
+const size      = ['S', 'M' , 'L' , 'XXL', 'XXXL'];
+const topBrands = ['IMARA', 'Biba' , 'Libas' , 'Manyavar', 'KISAH'];
+
+
+const FILTER_BRAND     = 'brand';
+const FILTER_SIZE      = 'size';
+const FILTER_IDEAL_FOR = 'ideal_for';
+const FILTER_MIN_PRICE = 'minPrice';
+const FILTER_MAX_PRICE = 'maxPrice';
+
+function FilterView({updateFilters}) {
+
+  const initialFilterState = {};
+  initialFilterState[FILTER_BRAND] = new Set([]);
+  initialFilterState[FILTER_IDEAL_FOR] = new Set([]);
+  initialFilterState[FILTER_SIZE] = new Set([]);
+  initialFilterState[FILTER_MIN_PRICE] = 0;
+  initialFilterState[FILTER_MAX_PRICE] = 1000;
+
+  const [filters, setFilters] = useState(initialFilterState);
+
+  function onFilterChange(e, filter, v){
+
+    if(filter === FILTER_MIN_PRICE || filter === FILTER_MAX_PRICE){
+      setFilters((prev) => ({ ...prev, [filter] : v}));
+    } else {
+      const {checked} = e.target;
+      if(checked){
+          setFilters((prev) => { prev[filter].add(v); return {...prev};});
```

Comment:
@rahulmeghwal If I'm reading this correctly, one concern might be with mutating the array vs using a more functional approach to return a new array populated with the new, correct items. I'm not sure if it's really a problem but I think it impacts reactivity?

Reply:
I agree, it can impact reactivity, since I was using Sets earlier and was returning a new object every time using previous values and new values, reactivity shouldn't be a problem here. However I changed the implementation to use arrays instead of sets and the code seems more readable and is perceived more functional now.